pylsp/plugins/rope_implementation.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2017-2020 Palantir Technologies, Inc.
+# Copyright 2021- Python Language Server Contributors.
+import logging
+import os
+from typing import Any
+
+from rope.base.project import Project
+from rope.base.resources import Resource
+from rope.contrib.findit import Location, find_implementations
+
+from pylsp import hookimpl, uris
+
+log = logging.getLogger(__name__)
+
+
+@hookimpl
+def pylsp_settings():
+    # Default to enabled (no reason not to)
+    return {"plugins": {"rope_implementation": {"enabled": True}}}
+
+
+@hookimpl
+def pylsp_implementations(config, workspace, document, position):
+    offset = document.offset_at_position(position)
+    rope_config = config.settings(document_path=document.path).get("rope", {})
+    rope_project = workspace._rope_project_builder(rope_config)
+    rope_resource = document._rope_resource(rope_config)
+
+    impls = find_implementations(rope_project, rope_resource, offset)
```

Comment:
Should it be wrapped in try-except to handle exceptions, as in:

https://github.com/python-lsp/python-lsp-server/blob/04fa3e59e82e05a43759f7d3b5bea2fa7a9b539b/pylsp/plugins/rope_completion.py#L59-L65